apps/remix-ide/src/app/udapp/run-tab.js

```diff
@@ -201,6 +201,31 @@ export class RunTab extends LibraryPlugin {
     this.renderDropdown(this.udappUI, this.fileManager, this.compilersArtefacts, this.config, this.editor, this.logCallback)
     this.renderRecorder(this.udappUI, this.fileManager, this.config, this.logCallback)
     this.renderRecorderCard()
+
+    this.on('manager', 'pluginDeactivated', profile => {
+      if (profile.kind === 'provider') this.blockchain.removeProvider(profile.name)
+    })
+    this.on('manager', 'pluginActivated', profile => {
+      if (profile.kind === 'provider') {
+        ((profile, app) => {
+          const web3Provider = {
+            sendAsync (payload, callback) {
+              console.log('wallet connect', payload)
+              app.call(profile.name, 'sendAsync', payload)
+                .then(result => {
+                  console.log('wallet connect', result)
+                  callback(null, result)
+                })
+                .catch(e => {
+                  console.log('wallet connect', e)
+                  callback(e)
+                })
+            }
+          }
+          this.blockchain.addProvider({ name: profile.displayName, provider: web3Provider })
+        })(profile, this)
```

Comment:
is this really necessary (() => {})() it feels like it could be removed, and use this instead of app

Reply:
`this` can be removed yes. 
`(() => {})()`  is to make sure the creation of the provider is encapsulated. (that the local var ` profile` doesn't get overridden in case the event callback is triggered multiple time.
But there might be better way to handle that.

Reply:
actually I had to keep `this` ...